src/terminatesteadystate.jl

```diff
@@ -0,0 +1,20 @@
+
+# Default test function
+# Terminate when all derivatives fall below a threshold, with early exit
+function allDerivPass(derivs, tol)
+    for der in derivs
+        der > tol && (return false)
+    end
+    return true
+end
+
+# Allow user-defined tolerances and test functions but use sensible defaults
+# test function must take derivatives as first argument, tolerance as second,
+#   return true/false
+function TerminateSteadyState(tol = 1e-8, test = allDerivPass)
+    condition = (u, t, integrator) -> test(integrator(t, Val{1}), tol)
```

Comment:
You can use `first(get_tmp_cache(integrator))` to get a cache array to write into and set this up with the in-place interpolation `integrator(out,t,Val{1})`. That'll speed up the common case, though we'll need to open an issue to make sure we handle the out of place (static array) case.

Reply:
Where would I create the cache? I mean the callback is created before it receives the integrator, no?